coding-the-legislation/reforms.md

```diff
@@ -1,104 +1,94 @@
+A reform that modifies a formula (such as our `income_tax_reform` example) is called a *structural reform*. It redefines the way a variable is calculated.
 
-```python
-def build_reform(tax_benefit_system):
-    Reform = reforms.make_reform(
-        key = 'change_formula',
-        name = u'Dummy reform with changed formula',
-        reference = tax_benefit_system,
-        )
-
-    @Reform.formula
-    class charges_deduc(formulas.SimpleFormulaColumn):
-        label = u"Charge déductibles always returning 999"
-        reference = charges_deductibles.charges_deduc
-        definition_period = YEAR
-        
-        def function(self, simulation, period):
-            return self.zeros() + 999
-
-    Reform.input_variable(
-        column = columns.BoolCol,
-        entity_class = entities.Menages,
-        name ='parisien',
-        label = u"Résidant à Paris au moins 3 ans dans les 5 dernières années",
-        )
-
-    reform = Reform()
-    return reform
-```
 
-To change the JSON data structure of the legislation in the reform, you call the `reform.modify_legislation_json` method which takes as a parameter a callback.
-This callback is a function you write in which the legislation JSON will be modified.
-It takes in parameter a copy of the legislation_json of the reference tax and benefit system that you can modify (do not make the copy yourself) and return.
+### Parametric reforms
+
+Unlike *structural reforms*, *parametric reforms* do not directly modify any formula: they apply changes to legislation parameters only.
 
-Here is an example:
+To modify the legislation parameters in the reform, you can call the method `self.modify_legislation_json`, which takes a function as a parameter.
+
+This function defines the modifications you want to apply to the legislation. It takes as a parameter a copy of the reference tax and benefit system `legislation_json`. You can then modify and return this `legislation_json`.
+
+For instance:
 
 ```python
 def modify_legislation_json(reference_legislation_json_copy):
-    reference_legislation_json_copy['children']['xxx']['values'][0]['value'] = 0
+    reference_legislation_json_copy['children']['minimum_wage']['values'][0]['value'] = 15
     return reference_legislation_json_copy
 
 
-def build_reform(tax_benefit_system):
-    Reform = reforms.make_reform(
-        key = 'new_legislation',
-        name = u'Dummy reform with new legislation',
-        reference = tax_benefit_system,
-        )
+class increase_minimum_wage(Reform):
+    name = u'Increase the minimum wage'
+
+    def apply(self):
+        self.modify_legislation_json(modifier_function = modify_legislation_json)
+```
+
+> Note that you have to know about the data structure of the legislation JSON to modify it.
+
+## Using a reform in Python
+
+Reforms can be applied in python with the following syntax:
+
+```py
+from openfica_some_country import CountryTaxBenefitSystem
+from some_other_package import income_tax_reform
```

Comment:
I agree focus is decreased, but I think it's ok since the elements that decrease focus have already been stated above, and that the change removes an additional question.